source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -181,14 +184,15 @@ void Filter::parseV2Header(char* buf) {
             hdr_addr_len - PROXY_PROTO_V2_ADDR_LEN_INET6, Network::Address::IpVersion::v6,
             std::make_shared<Network::Address::Ipv6Instance>(ra6),
             std::make_shared<Network::Address::Ipv6Instance>(la6)});
-        return;
+        return ProxyState::Done;
       }
     }
   }
-  throw EnvoyException("Unsupported command or address family or transport");
+  ENVOY_LOG(debug, "Unsupported command or address family or transport");
+  return ProxyState::UnsupportedCd;
 }
 
-void Filter::parseV1Header(char* buf, size_t len) {
+ProxyState Filter::parseV1Header(char* buf, size_t len) {
```

Comment:
The type signature for this in the header must be changed.

Reply:
I am not sure what change you are asking. Do you want the return-type of the method to be changed? Thanks!

Reply:
Yes, you changed it by making `void Filter::parserV1Header` to `ProxyState Filter::parseV1Header`. But you did not change that in the header file.

Reply:
Got it! fixed everything on the .h file!